### ISSUE TITLE
For mutable compile

### ISSUE THREAD

**Issue by mamiao** [mutable-org/mutable#1]
hi 
I am very interested for this project, and when I compile it using 
cmake -S . -B build/debug_shared \
> -G Ninja \
> -DCMAKE_C_COMPILER=clang \
> -DCMAKE_CXX_COMPILER=clang++ \
> -DCMAKE_BUILD_TYPE=Debug \
> -DBUILD_SHARED_LIBS=ON \
> -DENABLE_SANITIZERS=ON \
> -DENABLE_SANITY_FIELDS=ON \
> -DWITH_V8=ON

It will download the V8, but after it finish the download, the backend thread is still in the stage :
/bin/sh -c cd /Users/mezen/Downloads/master/mutable/third-party/v8 && fetch --force v8 || true && cd /Users/mezen/Downloads/master/mutable/third-party/v8/v8/ && git remote add mutable git@gitlab.cs.uni-saarland.de:bigdata/mutable/v8.git || true && cd /Users/mezen/Downloads/master/mutable/third-party/v8/v8/ && git fetch mutable && cd /Users/mezen/Downloads/master/mutable/third-party/v8/v8/ && git checkout adfc01872f43132c74bbd2182b127ad6f462f2c1 && /usr/local/Cellar/cmake/3.25.2/bin/cmake -E touch /Users/mezen/Downloads/master/mutable/build/debug_shared/third-party/src/V8-stamp/V8-download

and I can not see the V8-download file in the directory /Users/mezen/Downloads/master/mutable/build/debug_shared/third-party/src/V8-stamp/. So it means that in this stage it may fail

So could you please help me to fix it ?  Thanks

**Comment by ImmanuelHaffner** [comment]
Hi @mamiao!
Thanks for creating our **first issue on GitHub** 🥳 

V8 will be downloaded to `/path/to/mutable/third-party/v8/v8`.  Was this directory created? The directory should be the root of a Git repository. You can check with
```plain
$ git remote -v
mutable	git@gitlab.cs.uni-saarland.de:bigdata/mutable/v8.git (fetch)
mutable	git@gitlab.cs.uni-saarland.de:bigdata/mutable/v8.git (push)
origin	https://chromium.googlesource.com/v8/v8.git (fetch)
origin	https://chromium.googlesource.com/v8/v8.git (push)
```

**Comment by mamiao** [comment]
Also, you can see the  V8-stamp have two file which means the v8 install process, I think...

sh-3.2# ls /Users/mezen/Downloads/master/mutable/build/debug_shared/third-party/src/V8-stamp/
V8-custominfo.txt	V8-mkdir

**Comment by ImmanuelHaffner** [comment]
Thanks for checking. So far, so good. Then after configuration, what happens when compiling mu*t*able? The compilation step should download and compile our dependencies (the CMake `ExternalProject`s). Note, that there is no progress report on the command line for the compilation of V8. Compiling V8 for the first time, it can take 45-90 minutes (depending on your machine) 😞

**Comment by ImmanuelHaffner** [comment]
It would be great if you could provide the entire log of the compilation process.
That is all the output of `cmake --build ...`. Also, please use Markdown code blocks (```) to post the log here.

**Comment by mamiao** [comment]
> It would be great if you could provide the entire log of the compilation process. That is all the output of `cmake --build ...`. Also, please use Markdown code blocks (```) to post the log here.

ok~

**Comment by mamiao** [comment]
[mutable_compile.txt](https://github.com/mutable-org/mutable/files/10931028/mutable_compile.txt)
Thanks for your reply, I have uploaded the whole cmake log. Thanks

**Comment by ImmanuelHaffner** [comment]
Thanks a lot. I found the issue. The problem is that we used the `git@gitlab...` URL to our fork of V8 instead of the `https://...` URL. Hence, authentication fails and you cannot clone our V8 fork.
I am working on a fix. I'll let you know once this is solved.

**Comment by mamiao** [comment]
> Thanks a lot. I found the issue. The problem is that we used the `git@gitlab...` URL to our fork of V8 instead of the `https://...` URL. Hence, authentication fails and you cannot clone our V8 fork. I am working on a fix. I'll let you know once this is solved.

Thanks a lot !!!

**Comment by mamiao** [comment]
> Hey @mamiao, the problem should be fixed with commit [886a85b](https://github.com/mutable-org/mutable/commit/886a85bf011a3e63501fd270d36c08654459c06d). Please try again :)

👍     let me try。

**Comment by ImmanuelHaffner** [comment]
I'm glad it works!
